### ISSUE TITLE
`make install-deps` Inconsistencies

### ISSUE THREAD

**Issue by Ethan-DeBandi99** [Bears-R-Us/arkouda#1511]
After making Arrow a dependency due to import/export functionality for Parquet, we have seen some issues with environments being configured/updated properly. Additionally, we have seen issues with running `make install-deps` as well. Upon initial review, we have discovered a few things:
- Running `make install-deps` is not required if you install everything via conda/pip and have `Makefile.paths` and `LD_LIBRARY_PATH` configured properly.
- Our documentation does not fully highlight how to configure things in a clear manner.
- Versions in the `Makefile` are outdated. 

From here we need to do a few things.
- [x] Identify any issues when running `make install-deps` with a conda/pip env already configured. 
- [ ] Update documentation to indicate when `make install-deps` is needed to be clear. For developers/users running the server locally it should not be as their packages installed via conda/pip will work.
- [x] Update `Makefile` versions for dependencies.
- [ ] Add instructions on how to path `LD_LIBRARY_PATH` and `Makefile.paths` for each situation

*Note: This issue may be updated through its lifecycle if additional tasking is needed*

**Comment by Ethan-DeBandi99** [comment]
I went through the following steps to verify `make install-deps` is only needed if the users conda environment does not have `zeromq`, `hdf5`, or `arrow` installed (and is running locally). 

**Steps Taken:**
1. `unset LD_LIBRARY_PATH`. This is only to ensure that it is not being accessed to determine if it was needed (It is not)
2. Remove all contents of `Makefile.paths`. I believe if these are set to look at your conda env and pathed by `make install-deps` and versions differ, it will cause issues when running `make`
3. Updated `zmq` and `hdf5` versions to be installed by `make install-deps` to `zmq==4.3.4` and `hdf5==1.12.1`
4. `conda install cmake`. cmake is required for the `install-arrow` command.
5. `make install-deps`
6. `make`. Dependency checks pass and build completes successfully. 
7. Verified the the server turns on.

**Comment by Ethan-DeBandi99** [comment]
Tested configuration for local users and found the following:
- `Makefile.path` needs to be set to your env location. THIS SHOULD BE THE ONLY LINE IN THE FILE.
- `LD_LIBRARY_PATH` is not longer needed (at least in my testing. I will mess around with this more).
- The order of the channels in the `env.yml`s  as it is currently will break `arm64`. Flipping the order so `conda-forge` is first works for arm64 and x86. @mhmerrill I believe you had other config issues that we did not locate. 
- We need to require `pyarrow==7.0.0`.
- For mac users, brew install `zeromq` and `hdf5` is no longer required.
